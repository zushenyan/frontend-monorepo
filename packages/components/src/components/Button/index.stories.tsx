import React, { ReactElement } from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import Button from './index';

storiesOf('Button', module).add(
  'with text',
  (): ReactElement => <Button onClick={action('clicked')}>aaa</Button>,
);
