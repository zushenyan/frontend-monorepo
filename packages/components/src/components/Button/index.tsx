import styled from 'styled-components';

const Button = styled.button`
  background: none;
  border: 1px solid blue;
`;

export default Button;
