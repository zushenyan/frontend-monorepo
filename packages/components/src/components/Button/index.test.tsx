import React from 'react';
import renderer from 'react-test-renderer';
import initStoryshots from '@storybook/addon-storyshots';
import Button from './index';
import 'jest-styled-components';

initStoryshots({ framework: 'react', storyKindRegex: /^Button$/ });

describe('Button', (): void => {
  it('should work', (): void => {
    const onClick = jest.fn();
    const component = renderer.create(<Button onClick={onClick}>aaa</Button>);
    const instance = component.root;
    if (instance) {
      instance.props.onClick();
    }
    expect(onClick).toHaveBeenCalled();
  });
});
