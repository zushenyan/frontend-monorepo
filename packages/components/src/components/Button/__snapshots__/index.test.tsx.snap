// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Button with text 1`] = `
.c0 {
  background: none;
  border: 1px solid blue;
}

<button
  className="c0"
  onClick={[Function]}
>
  aaa
</button>
`;
